src/main/java/net/fabricmc/loom/util/GroovyXmlUtil.java

```diff
@@ -63,9 +65,19 @@ private static boolean isSameName(Object nodeName, String givenName) {
 			return ((QName) nodeName).matches(givenName);
 		}
 
+		// New groovy 3 (gradle 7) class
+		if (GradleSupport.IS_GRADLE_7_OR_NEWER && nodeName.getClass().getName().equals("groovy.namespace.QName")) {
```

Comment:
I think that you should be able to use `instanceof` here.
```suggestion
		if (GradleSupport.IS_GRADLE_7_OR_NEWER && nodeName instanceof groovy.namespace.QName) {
```

Reply:
That will fail when being ran on groovy 2.

Reply:
The short circuit should take care of it; no?

Reply:
No, it wont as far as I know. It will try to load the class when that method is first called.